doc/user/develop.rst

```diff
@@ -63,6 +60,128 @@ Custom functions that adjust time resolution can be loaded dynamically during mo
+
+---------------------
+Contribution workflow
+---------------------
+
+Have a bug fix or feature addition you'd like to see in the next stable release of Calliope? First, be sure to check out our list of `open <https://github.com/calliope-project/calliope/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen>`_ and `closed <https://github.com/calliope-project/calliope/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed>`_ issues to see whether this is something someone else has mentioned, or perhaps has even fixed. If it's there, you can add to the discussion, give it a thumbs up, or look to implement the change yourself. If it isn't there, then feel free to open your own issue, or you can head straight to implementing it. 
+
+Implementing a change
+---------------------
+
+When you want to change some part of Calliope, whether it is the software or the documentation, it's best to do it in a fork of the main Calliope project repository. You can find out more about how to fork a repository `here <https://help.github.com/articles/fork-a-repo/>`_. Your fork will be a duplicate of the Calliope master branch and can be 'cloned' to provide you with the repository on your own device::
+
+   $ git clone https://github.com/your_repository/calliope
+
+If you want the local version of your fork to be in the same folder as your local version of the main Calliope repository, then you just need to specify a new directory name::
+
+   $ git clone https://github.com/your_repository/calliope your_new_directory_name
+
+Following the instructions for :ref:`installing a development environment of Calliope <installing_dev>`, you can create an environment specific to this installation of Calliope. 
+
+In making changes to your local version, it's a good idea to create a branch first, to not have your master branch diverge from that of the main Calliope repository::
+
+   $ git branch new_fix_or_feature
+
+Then, 'checkout' the branch so that the folder contents are specific to that branch::
+
+   $ git checkout new_fix_or_feature
+
+Finally, push the branch online, so it's existence is also in your remote fork of the Calliope repository (you'll find it in the dropdown list of branches at https://github.com/your_repository/calliope)::
+
+   $ git push -u origin feature_branch_name
+
+Now the files in your local directory can be edited with complete freedom. Once you have made the necessary changes, you'll need to test that they don't break anything. This can be done easily by changing to your fork directory from the terminal / command line and running `pytest <https://docs.pytest.org/en/latest/index.html>`_ (make sure you're in your calliope fork conda environment and you have pytest installed: `conda install pytest`). Any change you make should also be covered by a test. Add it into the relevant test file, making sure the function starts with 'test\_'. Since the whole test suite takes ~25 minutes to run, you can run specific tests, such as those you add in::
```

Comment:
```suggestion
Now the files in your local directory can be edited with complete freedom. Once you have made the necessary changes, you'll need to test that they don't break anything. This can be done easily by changing to the directory into which you cloned your fork using the terminal / command line, and running `pytest <https://docs.pytest.org/en/latest/index.html>`_ (make sure you have activated the conda environment and you have pytest installed: `conda install pytest`). Any change you make should also be covered by a test. Add it into the relevant test file, making sure the function starts with 'test\_'. Since the whole test suite takes ~25 minutes to run, you can run specific tests, such as those you add in::
```